program/src/processor.rs

```diff
@@ -44,78 +45,83 @@ fn calculate_proposal_vote_threshold(stake: u64, total_stake: u64) -> Result<u64
 
 fn get_stake_checked(
     authority_key: &Pubkey,
+    stake_config_address: &Pubkey,
     stake_info: &AccountInfo,
-    stake_config_info: &AccountInfo,
-) -> Result<(u64, u64), ProgramError> {
-    // Ensure the stake account is owned by the Paladin Stake program.
-    if stake_info.owner != &paladin_stake_program::id() {
-        return Err(ProgramError::InvalidAccountOwner);
+) -> Result<u64, ProgramError> {
+    let data = stake_info.try_borrow_data()?;
+    let state =
+        bytemuck::try_from_bytes::<Stake>(&data).map_err(|_| ProgramError::InvalidAccountData)?;
+
+    // Ensure the stake account belongs to the authority.
+    if state.authority != *authority_key {
+        return Err(ProgramError::IncorrectAuthority);
     }
 
-    let stake = {
-        let data = stake_info.try_borrow_data()?;
-        let state = bytemuck::try_from_bytes::<Stake>(&data)
-            .map_err(|_| ProgramError::InvalidAccountData)?;
-
-        // Ensure the stake account is initialized.
-        if !state.is_initialized() {
-            return Err(ProgramError::UninitializedAccount);
-        }
-
-        // Ensure the stake account belongs to the authority.
-        if state.authority != *authority_key {
-            return Err(ProgramError::IncorrectAuthority);
-        }
+    // Ensure the stake account has the correct address derived from the
+    // validator vote account and the stake config account.
+    if stake_info.key
+        != &find_stake_pda(
+            &state.validator_vote,
+            stake_config_address,
+            &paladin_stake_program::id(),
+        )
+        .0
+    {
+        return Err(PaladinGovernanceError::StakeConfigMismatch.into());
+    }
 
-        // Ensure the stake account has the correct address derived from the
-        // validator vote account and the stake config account.
-        if stake_info.key
-            != &find_stake_pda(
-                &state.validator_vote,
-                stake_config_info.key,
-                &paladin_stake_program::id(),
-            )
-            .0
-        {
-            return Err(PaladinGovernanceError::StakeConfigMismatch.into());
-        }
+    Ok(state.amount)
+}
 
-        state.amount
-    };
+fn get_total_stake_checked(
+    _governance_config_address: &Pubkey,
+    stake_config_info: &AccountInfo,
+) -> Result<u64, ProgramError> {
+    let data = stake_config_info.try_borrow_data()?;
+    let state = bytemuck::try_from_bytes::<StakeConfig>(&data)
+        .map_err(|_| ProgramError::InvalidAccountData)?;
 
     // TODO: This will probably require the governance config account, once
     // the program can support multiple governance configs.
     // Something like storing the stake config address on the governance config
     // should do it.
 
+    Ok(state.token_amount_delegated)
+}
+
+fn check_stake_config_exists(stake_config_info: &AccountInfo) -> ProgramResult {
```

Comment:
Same with this: do you want to call `check_stake_config_exists` in `get_total_stake_checked` to avoid accidental misses?

Reply:
I'm pretty sure I end up evicting the `get_total_stake_checked` helper somewhere along the line, but to be safe I'll add this suggestion!

Reply:
You're right, it's removed in the next PR, so up to you!